File: android/src/main/java/expo/modules/rexpayreactnative/RexpayReactNativeModule.kt
package expo.modules.webview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class RexpayReactNativeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("RexpayReactNativeView")

    View(RexpayReactNativeView::class) {
      Events("onLoad", "onNavigationStateChange")

      Prop("url") { view: ExpoWebView, url: URL? ->
        view.webView.loadUrl(url.toString())
      }
    }
  }
}

File: ios/RexpayReactNativeModule.swift
import ExpoModulesCore

public class RexpayReactNativeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("RexpayReactNativeView")

    View(RexpayReactNativeView.self) {
      Events("onLoad", "onNavigationStateChange")

      Prop("url") { (view, url: URL) in
        if view.webView.url != url {
          let urlRequest = URLRequest(url: url)
          view.webView.load(urlRequest)
        }
      }
    }
  }
}
